### PR TITLE
Fix the scheduler to ignore terminated pods.

### PR DESCRIPTION
**Pull request by brendandburns**
@lavalamp @bgrant0607 @dchen1107

### CODE REVIEW

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 1]
```diff
@@ -369,6 +383,7 @@ func MapPodsToMachines(lister algorithm.PodLister) (map[string][]*api.Pod, error
 	if err != nil {
 		return map[string][]*api.Pod{}, err
 	}
+	pods = filterNonRunningPods(pods)
```
Comment (lavalamp) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: Should this be pushed down into the lister? the lister already ignores some things, IIRC.

**Comment by lavalamp** [comment]
Other than that, this LGTM.

**Comment by lavalamp** [comment]
LGTM

**Comment by dchen1107** [comment]
LGTM

**Comment by brendandburns** [comment]
Closes https://github.com/GoogleCloudPlatform/kubernetes/issues/9601

**Comment by davidopp** [comment]
LGTM, thanks for fixing!

**Pull request merged**